### PR TITLE
Support named parameters for PRAGMA statements

### PR DESCRIPTION
**Pull request by lnkuiper**
Support e.g. `PRAGMA some_function(1, x=10)` instead of `PRAGMA some_function(1, 10)` so certain parameters can have default values, and it is more clear which parameters you are setting.

TableFunctions already support this. Initially I tried to unify these, but ran into a bunch of problems, because of differences between how PragmaStatement and TableRef store their parameters.

Now they are different in the parser, but under function TableFunction and PragmaFunction are derived from the same base class, SimpleNamedParameterFunction.

There are no PRAGMA's that currently need this, but I have plans! Also, there are no tests, but I did define a useless PRAGMA to test.

### CODE REVIEW

File: src/function/pragma/pragma_queries.cpp [hunk 1]
```diff
@@ -4,35 +4,40 @@
...
 }
 
-string pragma_collations(ClientContext &context, vector<Value> parameters) {
+string pragma_collations(ClientContext &context, vector<Value> parameters,
+                         unordered_map<string, Value> named_parameters) {
 	return "SELECT * FROM pragma_collations() ORDER BY 1";
 }
 
-string pragma_show(ClientContext &context, vector<Value> parameters) {
+string pragma_show(ClientContext &context, vector<Value> parameters, unordered_map<string, Value> named_parameters) {
```
Comment (Mytherin) [inline src/function/pragma/pragma_queries.cpp:27]: Maybe an idea to pack the parameters into a struct, e.g.

```cpp
struct FunctionParameters {
   vector<Value> values;
   unordered_map<string, Value> named_parameters;
};
```

That can also be used in the table functions.

File: src/planner/binder/statement/bind_pragma.cpp [hunk 2]
```diff
@@ -20,6 +20,27 @@ BoundStatement Binder::Bind(PragmaStatement &stmt) {
 		throw BinderException("PRAGMA function does not have a function specified");
 	}
 
+	QueryErrorContext error_context(root_statement, stmt.stmt_location);
+	// bind and check named params
+	for (auto &kv : stmt.info->named_parameters) {
+		auto entry = bound_function.named_parameters.find(kv.first);
+		if (entry == bound_function.named_parameters.end()) {
+			// create a list of named parameters for the error
+			string named_params;
```
Comment (Mytherin) [inline src/planner/binder/statement/bind_pragma.cpp:29]: Can this be unified with the table function printing code?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! I think it looks good; I would just request unifying the printing of the parameters, and perhaps introducing a struct to hold them

File: src/function/pragma/pragma_queries.cpp (hunk 1 above)
Comment (lnkuiper) [reply src/function/pragma/pragma_queries.cpp:27]: Good idea, this makes it look more clean

File: src/planner/binder/statement/bind_pragma.cpp (hunk 2 above)
Comment (lnkuiper) [reply src/planner/binder/statement/bind_pragma.cpp:29]: Sure, this is pretty much a copy-paste of some code in bind_table_function.cpp.

Where should I place this unified piece of code? I'd say function.cpp would be nice, but there's some stuff related to the binder in this piece of code, so perhaps somewhere in the binder?

**Comment by lnkuiper** [comment]
I made a few oopsies when implementing the feedback, but everything works in the travis of my fork now. I see that 2 jobs fail in travis here, but I don't exactly understand what's wrong with them by looking at the output. Any ideas?

**Comment by Mytherin** [comment]
The failures look rather innocent, there is a problem with the cibuildwheel for Windows which causes the failure there. The other test sometimes fails randomly because Postgres doesn’t start up correctly; no clue about why but it has nothing to do with your branch. I will rerun the Postgres one and have another look at the code, after that I think it’s ready to merge.

**Comment by lnkuiper** [comment]
All tests now pass, seems like the efforts to fix Travis have worked!

**Comment by Mytherin** [comment]
Great! Then I think this can be merged :)

**Pull request merged**